tests/infra/jwt.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import infra.crypto
+import base64
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from http import HTTPStatus
+import ssl
+import threading
+from contextlib import AbstractContextManager
+import tempfile
+import json
+import time
+from loguru import logger as LOG
+
+
+def make_authn_bearer_header(jwt):
+    return {"authorization": "Bearer " + jwt}
+
+
+class MyHTTPRequestHandler(BaseHTTPRequestHandler):
+    def __init__(self, openid_server, *args):
+        self.openid_server = openid_server
+        BaseHTTPRequestHandler.__init__(self, *args)
+
+    def do_GET(self):
+        routes = {
+            "/.well-known/openid-configuration": self.openid_server.metadata,
+            "/keys": self.openid_server.jwks,
+        }
+        body = routes.get(self.path)
+        if body is None:
+            self.send_error(HTTPStatus.NOT_FOUND)
+            return
+        body = json.dumps(body).encode()
+        self.send_response(HTTPStatus.OK)
+        self.send_header("Content-Length", str(len(body)))
+        self.end_headers()
+        self.wfile.write(body)
+
+    def log_message(self, fmt, *args):  # pylint: disable=arguments-differ
+        LOG.debug(f"OpenIDProviderServer: {fmt % args}")
+
+
+class OpenIDProviderServer(AbstractContextManager):
+    def __init__(self, port: int, tls_key_pem: str, tls_cert_pem: str, jwks: dict):
+        self.host = "localhost"
+        self.port = port
+        self.jwks = jwks
+        self.tls_key_pem = tls_key_pem
+        self.tls_cert_pem = tls_cert_pem
+        self.self_ = self
+        self.bind_port = None
+        self.started = False
+        self.start(self.port)
+
+    def start(self, port):
```

Comment:
What's the difference between port and bind_port?

Reply:
To avoid port collisions, we now ask to bind to `0` by default. Once this is done, we get the port the server actually bound to (`bind_port`) and use that to register the issuer URL to CCF.